### PR TITLE
Polyhedron demo: add qtscript support to Mesh_3 plugin

### PR DESCRIPTION
**Pull request by lrineau**
## Summary of Changes

Restore in the CGAL 3D demo the possibility to use Qt Script with the mesh_3_plugin. Inspired by the branch `Mesh_3-experimental-GF`.

## Release Management

* Affected package(s): Polyhedron demo, Mesh_3
* License and copyright ownership: maintenance by GeometryFactory

### CODE REVIEW

File: Mesh_3/include/CGAL/Mesh_3/Protect_edges_sizing_field.h [hunk 1]
```diff
@@ -444,7 +444,13 @@ class Protect_edges_sizing_field
       if(dim == 0) msg << "corner (";
       else msg << "point (";
       msg << p << ")";
-      CGAL_error_msg(msg.str().c_str());
+      CGAL_error_msg(([this, str = msg.str()](){
+        CGAL_USE(this);
+#if CGAL_MESH_3_PROTECTION_DEBUG & 4
```
Comment (sloriot) [inline Mesh_3/include/CGAL/Mesh_3/Protect_edges_sizing_field.h]: @maxGimeno could you please check why this results in the following [error](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.1-Ic-100/Generator/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Release-64bits.gz)

Comment (lrineau) [reply Mesh_3/include/CGAL/Mesh_3/Protect_edges_sizing_field.h]: My bad. One should not use preprocessor directive inside the arguments of a macro. That is not portable.

Comment (lrineau) [reply Mesh_3/include/CGAL/Mesh_3/Protect_edges_sizing_field.h]: Fixed by https://github.com/CGAL/cgal/pull/4074/commits/6d215c776bde45f59f3afb386aea4372a0777a2f.

**Comment by maxGimeno** [comment]
I still see https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.1-Ic-106/Generator/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Debug-64bits.gz

**Comment by lrineau** [comment]
> I still see [cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.1-Ic-106/Generator/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Debug-64bits.gz](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.1-Ic-106/Generator/TestReport_afabri_x64_Cygwin-Windows10_MSVC2017-Debug-64bits.gz)

Should be fixed by https://github.com/CGAL/cgal/pull/4074/commits/e6a42bb597488cb3fe45616ebb4fa0ef47696da4.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-108.shtml

**Pull request merged**